system-tests/mpi/osu/osu_check.py

```diff
@@ -0,0 +1,179 @@
+            'cpu': {
+                'lumi:small': {
+                    'latency': (1.8, None, 0.80, 'us')
+                }
+            },
+            'rocm': {
+                'lumi:gpu': {
+                    'latency': (3.75, None, 0.50, 'us')
+                }
+            }
+        }
+    }
+
+    @run_after('init')
+    def setup_per_build_type(self):
+        build_type = self.osu_binaries.build_type
+        if build_type == 'rocm':
+            self.valid_systems = ['lumi:gpu']
+            self.device_buffers = 'rocm'
+            self.num_gpus_per_node = 1
+            self.executable_opts = ['-c', '-d', 'rocm', 'D', 'D']
+            self.valid_prog_environs = ['builtin-hip']
+            self.variables = {'MPICH_GPU_SUPPORT_ENABLED': '1'} 
+        else:
+            self.valid_systems = ['lumi:small']
+            self.valid_prog_environs = ['cpeGNU', 'cpeCray']
+            self.executable_opts = ['-c']
+
+        with contextlib.suppress(KeyError):
+            self.reference = self.allref[self.benchmark_info[0]][build_type]
+
+    @run_before('run')
+    def add_exec_prefix(self):
+        build_type = self.osu_binaries.build_type
+        bench_path = self.benchmark_info[0].replace('.', '/')
+        # update directory structure
+        self.executable = os.path.join(self.osu_binaries.stagedir,
+                                       self.osu_binaries.build_prefix,
+                                       'c',
+                                       bench_path)
+        if build_type == 'rocm':
+            self.executable = os.path.join(self.osu_binaries.stagedir, self.osu_binaries.build_prefix,
+                                           'map_rank_to_gpu ') + self.executable
+
+    @sanity_function
+    def validate_test(self):
+        # with validation
+        return sn.assert_found(rf'^{self.message_size}.*Pass', self.stdout)
+
+
+@rfm.simple_test
+class lumi_osu_collective_check(lumi_osu_benchmarks):
+    benchmark_info = parameter([
+        #('mpi.collective.osu_alltoall', 'latency'),
+        ('mpi.collective.osu_allreduce', 'latency'),
+    ], fmt=lambda x: x[0], loggable=True)
+    num_nodes = parameter([1, 2])
+    use_multithreading = False
+    valid_systems = ['lumi:small', 'lumi:gpu'] 
+    valid_prog_environs = ['cpeGNU', 'cpeCray', 'builtin-hip']
+    osu_binaries = fixture(lumi_build_osu_benchmarks, scope='environment')
+    allref = {
+        'mpi.collective.osu_allreduce': {
+            1: {
+                'lumi:small': {
+                    'latency': (10.0, None, 0.1, 'us')
+                }
+            },
+            2: {
+                'lumi:small': {
+                    'latency': (7.25, None, 0.05, 'us')
+                }
+            },
+        },
+    }
+
+    @run_after('init')
+    def setup_by_scale(self):
+        with contextlib.suppress(KeyError):
+            self.reference = self.allref[self.num_nodes]
+
+    @run_after('init')
+    def setup_per_build_type(self):
```

Comment:
Maybe most of this function can be moved to `lumi_osu_benchmarks` and be used both in the pt2pt and collective tests, each one then addeding the particular variables needed in each case?